File: app/src/main/java/com/mgaetan89/showsrage/network/OmDbApi.java
package com.mgaetan89.showsrage.network;

import com.mgaetan89.showsrage.model.OmDbEpisode;
import com.mgaetan89.showsrage.model.Serie;

import retrofit.Callback;
import retrofit.http.GET;
import retrofit.http.Query;

public interface OmDbApi {
	@GET("/?plot=full")
	void getEpisodeByImDbId(@Query("i") String imDbId, @Query("Season") int season, @Query("Episode") int episode, Callback<OmDbEpisode> callback);

	@GET("/?plot=full")
	void getEpisodeByTitle(@Query("t") String title, @Query("Season") int season, @Query("Episode") int episode, Callback<OmDbEpisode> callback);

	@GET("/?plot=full")
	void getShow(@Query("i") String imDbId, Callback<Serie> callback);
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/network/OmDbApi.kt
package com.mgaetan89.showsrage.network

import com.mgaetan89.showsrage.model.OmDbEpisode
import com.mgaetan89.showsrage.model.Serie

import retrofit.Callback
import retrofit.http.GET
import retrofit.http.Query

interface OmDbApi {
	@GET("/?plot=full")
	fun getEpisodeByImDbId(@Query("i") imDbId: String, @Query("Season") season: Int, @Query("Episode") episode: Int, callback: Callback<OmDbEpisode>)

	@GET("/?plot=full")
	fun getEpisodeByTitle(@Query("t") title: String, @Query("Season") season: Int, @Query("Episode") episode: Int, callback: Callback<OmDbEpisode>)

	@GET("/?plot=full")
	fun getShow(@Query("i") imDbId: String, callback: Callback<Serie>)
}
